Sources/XCBuildSupport/PIFBuilder.swift

```diff
@@ -570,7 +571,7 @@ final class PackagePIFProjectBuilder: PIFProjectBuilder {
         addSources(target.sources, to: pifTarget)
 
         // Handle the target's dependencies (but don't link against them).
-        for dependency in target.recursivePackageDependencies(satisfying: parameters.buildEnvironment) {
+        for dependency in target.dependencies {
```

Comment:
Have you tested that we don't need recursive dependencies here?

Reply:
Perhaps we should wait until integration tests are run with CI to fix that.

Reply:
I tested some but, yeah, we should get integration tests up and running.

Reply:
Unfortunately, we don't have the latest Xcode on the CI yet so I'll test locally for now. I am pretty confident that this works.

Reply:
Tested this locally.